website/docs/plugin/framework/functions/implementation.mdx

```diff
@@ -0,0 +1,252 @@
+
+If the logic needs to return [warning or error diagnostics](/terraform/plugin/framework/diagnostics), they can be added into the [`function.RunResponse.Diagnostics` field](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/function#RunResponse.Diagnostics).
+
+### Reading Argument Data
+
+The framework supports two methodologies for reading argument data from the [`function.RunRequest.Arguments` field](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/function#RunRequest.Arguments), which is of the [`function.ArgumentsData` type](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/function#ArgumentsData).
+
+The first option is using the [`(function.ArgumentsData).Get()` method](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/function#ArgumentsData.Get) to read all arguments at once. The framework will return errors if the number and types of target variables does not match the argument data.
+
+In this example, the parameters are defined as a boolean and string which are read into Go built-in `bool` and `string` variables since they do not opt into null or unknown value handling:
+
+```go
+func (f *ExampleFunction) Definition(ctx context.Context, req function.DefinitionRequest, resp *function.DefinitionResponse) {
+    resp.Definition = function.Definition{
+        // ... other fields ...
+        Parameters: []function.Parameter{
+            function.BoolParameter{},
+            function.StringParameter{},
+        },
+    }
+}
+
+func (f *ExampleFunction) Run(ctx context.Context, req function.RunRequest, resp *function.RunResponse) {
+    var boolArg bool
+    var stringArg string
+
+    resp.Diagnostics.Append(req.Arguments.Get(ctx, &boolArg, &stringArg)...)
+
+    // ... other logic ...
+}
+```
+
+The second option is using [`(function.ArgumentsData).GetArgument()` method](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/function#ArgumentsData.GetArgument) to read individual arguments. The framework will return errors if the argument position does not exist or if the type of the target variable does not match the argument data.
+
+In this example, the parameters are defined as a boolean and string and the first argument is read into a Go built-in `bool` variable since it does not opt into null or unknown value handling:
+
+```go
+func (f *ExampleFunction) Definition(ctx context.Context, req function.DefinitionRequest, resp *function.DefinitionResponse) {
+    resp.Definition = function.Definition{
+        // ... other fields ...
+        Parameters: []function.Parameter{
+            function.BoolParameter{},
+            function.StringParameter{},
+        },
+    }
+}
+
+func (f *ExampleFunction) Run(ctx context.Context, req function.RunRequest, resp *function.RunResponse) {
+    var boolArg bool
+
+    resp.Diagnostics.Append(req.Arguments.GetArgument(ctx, 0, &boolArg)...)
+
+    // ... other logic ...
+}
+```
+
```

Comment:
Should we provide a section with examples on reading variadic arguments (i.e. with a list)

Reply:
👍 Indeed. Hope the page content isn't getting too long.

Reply:
Took a swing at this here: https://github.com/hashicorp/terraform-plugin-framework/pull/889/commits/c4a1246702de3e234e03f5104895e73832ca3b3a -- please let me know what you think.

Reply:
Just one word removal, but the content looks great, thanks!